src/App.js

```diff
@@ -28,13 +36,25 @@ class App extends Component {
         this._fetchData();
     }
 
-    handleSpeciesChange(species) {
-        this.setState({species});
-        this._fetchStudies();
+    handleSpeciesChange(selectedSpecies) {
+        const assemblyList = this._resolveAssemblyList(this.state.speciesAssemblyMap, selectedSpecies);
+        const selectedAssembly = selectedSpecies + "_" + assemblyList[0].assemblyCode
+        this.setState({
+            assemblyList: assemblyList,
+            selectedSpecies: selectedSpecies,
+            selectedAssembly: selectedAssembly,
+            location: this.assemblyPositionMap[selectedAssembly]
+        }, this._fetchStudies);
+    }
+
+    handleAssemblyChange(selectedAssembly) {
+        this.setState({
+            selectedAssembly: selectedAssembly, location: this.assemblyPositionMap[selectedAssembly]
```

Comment:
Also I think this should have a default value, for assemblies not in the beautiful hardcoded map